src/Avalonia.Base/Controls/INotSharedDeferredContent.cs

```diff
@@ -0,0 +1,8 @@
+﻿namespace Avalonia.Controls;
+
+/// <summary>
+/// Represents a not shared deferred content.
+/// </summary>
+internal interface INotSharedDeferredContent: IDeferredContent
```

Comment:
Now that's it's not used at all in public API, I think you can completely remove it and simply check for `NotSharedDeferredItem` in the `ResourceDictionary` instead.